File: src/test/scala/workshop/mandelbrot/PixelSolverTester.scala
package workshop.mandelbrot

import org.scalatest.FunSuite
import spinal.core._
import spinal.core.sim._
import workshop.common.WorkshopSimConfig

//Run this scala test to generate and check that your RTL work correctly
class PixelSolverTester extends FunSuite{
  var compiled: SimCompiled[PixelSolver] = null

  test("compile") {
    compiled = WorkshopSimConfig().compile(
      PixelSolver(
        g = PixelSolverGenerics(
          fixAmplitude = 7,
          fixResolution = -20,
          iterationLimit = 15
        )
      )
    )
  }

  test("testbench") {
    compiled.doSimUntilVoid(seed = 42){dut =>
      dut.clockDomain.forkStimulus(10)
      PixelSolverChecker(dut.io.cmd, dut.io.rsp, dut.clockDomain)
    }
  }
}

File: src/test/python/workshop/mandelbrot/PixelSolverTester/PixelSolverTester.py
import random

import cocotb

from cocotblib.misc import SimulationTimeout
from workshop.mandelbrot.common import pixelSolverTester


@cocotb.test()
def test1(dut):
    cocotb.fork(SimulationTimeout(1000 * 80000))
    yield pixelSolverTester(dut)
